### PR TITLE
Migrate bulk IOs to accept AttributeFamilyDescriptor + fix shading

### PR DESCRIPTION
**Pull request by LesTR**
* Migrage io-s3, io-gcloud and io-bulkfs  api to be created via AttributeFamilyDescriptor
* Added setup method into `FileFormat`
* `org.apache.http.` needs to be shaded in io-s3
* Implement returning schema from `AlwaysFailSchemeParser`

### CODE REVIEW

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java [hunk 1]
```diff
@@ -56,26 +55,38 @@
...
+       * be able to override it from tests
+       */
+      cfg = new HashMap<>(familyDescriptor.getCfg());
+    }
     return Collections.unmodifiableMap(cfg);
   }
 
   @VisibleForTesting
   void setCfg(String key, Object value) {
+    getCfg();
```
Comment (je-ik) [inline direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java:80]: Is this OK? This introduces a somewhat unexpected contract:
```java
 accessor.setCfg(...);
 cloned = SerializableUtils.clone(accessor);
 cloned.getCfg(...); // the set configuration will be missing here
 ```

**Review by je-ik — COMMENTED** [review]
+0
I'm not sure if fixing the serializability contract is worth it. If we can fix that easily, we might do that, the setCfg is `@VisibleForTesting`, so that is not that huge issue.
If fixing that would be complicated, we can merge this as it is.

File: direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java (hunk 1 above)
Comment (LesTR) [reply direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java]: Reverted back, I think fixing Serializable contract doesn't make sense here and we can probably remove it later.

**Review by je-ik — APPROVED** [review]
LGTM! :+1:

**Pull request merged**